python/cudf/cudf/core/frame.py

```diff
@@ -582,16 +574,35 @@ def equals(self, other, **kwargs):
         else:
             return self._index.equals(other._index)
 
-    def _explode(self, explode_column_num: int, index: Optional[cudf.Index]):
-        if index is not None:
-            explode_column_num += index.nlevels
-        res_tbl = libcudf.lists.explode_outer(
-            cudf._lib.table.Table(self._data, index=index), explode_column_num
-        )
+    def _explode(self, explode_column: Any, ignore_index: bool):
+        """Helper function for `explode` in Series and Dataframe.
+        if the designated column to explode is non-nested, a copy
+        of the frame is returned. Otherwise, if ignore_index is
+        set, the original index is not exploded and will use
+        a `RangeIndex` instead.
+        """
+        if not is_list_dtype(self._data[explode_column].dtype):
+            copy = self.copy()
+            if ignore_index:
+                copy._index = cudf.RangeIndex(copy._num_rows)
+            return copy
 
+        explode_column_num = self._column_names.index(explode_column)
+        if ignore_index:
+            tmp_index, self._index = self._index, None
```

Comment:
Mutating `self._index` is not safe here, if something throws we can end up having messed with the input object.

Why do we need `tmp_index` here?

Reply:
I'm hoping to temporarily remove the `_index` field from the object so that down at the `Table` level the index is not fed to libcudf.

Reply:
I think the way we typically handle this is by passing the `ignore_index` option down into the Cython and using it to control whether we call `Table.view()` vs `Table.data_view()`: https://github.com/rapidsai/cudf/blob/branch-0.19/python/cudf/cudf/_lib/copying.pyx#L147-L150

Reply:
At this point, I can think of 2 solutions:
One is to create a shallow copy then drop index as needed:
```python
copy = self.copy(deep=False)
copy._index = None if ignore_index else copy._index
...
```
Second is to construct a new `Table` (a simplified shallow copy)
```python
tbl = Table(self._data, index=self._index if ignore_index is not None else None)
...
```